src/ansys/geometry/core/shapes/curves/trimmed_curve.py

```diff
@@ -154,6 +156,90 @@ def intersect_curve(self, other: "TrimmedCurve") -> list[Point3D]:
             for point in res.points
         ]
 
+    def transformed_copy(self, matrix: Matrix44) -> "TrimmedCurve":
+        """Return a copy of the trimmed curve transformed by the given matrix.
+
+        Parameters
+        ----------
+        matrix : Matrix44
+            Transformation matrix to apply to the curve.
+
+        Returns
+        -------
+        TrimmedCurve
+            A new trimmed curve with the transformation applied.
+        """
+        transformed_geometry = self.geometry.transformed_copy(matrix)
+        transformed_start = self.start.transform(matrix)
+        transformed_end = self.end.transform(matrix)
+
+        return TrimmedCurve(
+            transformed_geometry,
+            transformed_start,
+            transformed_end,
+            self.interval,
+            self.length,
+        )
+
+    def translate(self, direction: Vector3D, distance: Real | Quantity | Distance) -> None:
+        """Translate the trimmed curve by a given vector and distance.
+
+        Parameters
+        ----------
+        direction : Vector3D
+            Direction of translation.
+        distance : Real | Quantity | Distance
+            Distance to translate.
+        """
+        distance = distance if isinstance(distance, Distance) else Distance(distance)
+        translation_matrix = Matrix44.create_translation(direction * distance.value.m)
+
+        translated_copy = self.transformed_copy(translation_matrix)
+
+        # Update the current instance with the translated copy
+        self._geometry = translated_copy.geometry
+        self._start = translated_copy.start
+        self._end = translated_copy.end
+        self._length = translated_copy.length
+        self._interval = translated_copy.interval
+
+    def rotate(self, origin: Point3D, axis: Vector3D, angle: Real | Quantity | Angle) -> None:
```

Comment:
Same here for Axis I believe. Otherwise the transformation will include an undesired magnitude